src/front/wgsl.rs

```diff
@@ -503,6 +522,45 @@ impl Parser {
         }
     }
 
+    fn parse_function_call<'a>(
+        &mut self,
+        lexer: &mut Lexer<'a>,
+        mut ctx: ExpressionContext<'a, '_, '_>,
+    ) -> Result<crate::Expression, Error<'a>> {
```

Comment:
the error model of this function is weird: it tries to return parsing errors, but they never used by the callers
Perhaps, it should instead just return an `Option<>`?

Reply:
You're right, this should return `Option`.